src/Uno.UI/UI/Xaml/Internal/DependencyPropertyHelper.cs

```diff
@@ -0,0 +1,209 @@
+#nullable enable
+
+using Microsoft.UI.Xaml;
+using System;
+using System.Collections.Generic;
+using System.Diagnostics.CodeAnalysis;
+using Microsoft.UI.Xaml.Controls;
+
+namespace Uno.UI.Xaml.Core;
+
+/// <summary>
+/// The goal of this class is to provide a set of helper methods to work with <see cref="DependencyProperty"/> from
+/// external projects.
+/// </summary>
+/// <remarks>
+/// A small reflection is still required to access the class because those using it needs to know what they are doing.
+/// </remarks>
+internal static class DependencyPropertyHelper
+{
+	/// <summary>
+	/// Get the default value of a <see cref="DependencyProperty"/> for the owner type. 
+	/// </summary>
+	/// <remarks>
+	/// This is the value defined in the metadata of the property, which _may_ be overriden
+	/// using the internal DependencyProperty.OverrideMetadata() method on a per-type basis
+	/// -- that's why the type parameter is required.
+	///
+	/// A classic example of that is the Button.VerticalAlignmentProperty which is overridden
+	/// to be VerticalAlignment.Center, while the default value is VerticalAlignment.Stretch
+	/// from the base class.
+	///
+	/// This overload will return the default value for the owner type of the property - where
+	/// the value is originally defined.
+	/// </remarks>
+	public static object? GetDefaultValue(DependencyProperty dependencyProperty)
+		=> dependencyProperty.GetMetadata(dependencyProperty.OwnerType)?.DefaultValue;
+
+	/// <summary>
+	/// Get a reference to <see cref="DependencyProperty"/> by its name and owner type.
+	/// </summary>
+	/// <remarks>
+	/// The name will match the name of the property as defined when the property is registered.
+	///
+	/// There is usually NO "Property" suffix on that name since it's the name that is used in XAML.
+	///
+	/// The name is case-sensitive.
+	/// </remarks>
+	public static DependencyProperty? GetDependencyPropertyByName(Type ownerType, string propertyName)
+		=> DependencyProperty.GetProperty(ownerType, propertyName);
+
+	/// <summary>
+	/// Get a reference to <see cref="DependencyProperty"/> by its name on the given type.
+	/// </summary>
+	/// <remarks>
+	/// The name will match the name of the property as defined when the property is registered.
+	///
+	/// There is usually NO "Property" suffix on that name since it's the name that is used in XAML.
+	///
+	/// The name is case-sensitive.
+	/// </remarks>
+	public static DependencyProperty? GetDependencyPropertyByName<T>(string propertyName)
+		where T : DependencyObject
+		=> GetDependencyPropertyByName(typeof(T), propertyName);
+
+	/// <summary>
+	/// Get all the <see cref="DependencyProperty"/> defined for a given type. 
+	/// </summary>
+	public static IReadOnlyCollection<DependencyProperty>? GetDependencyPropertiesForType<T>()
+		where T : DependencyObject
+		=> DependencyProperty.GetPropertiesForType(typeof(T));
+
```

Comment:
This will be removed with https://github.com/unoplatform/uno/pull/18089

Reply:
Let's not include it with this PR for now at least and we can discuss it afterwards